### ISSUE TITLE
Grid Search Error: Error in FUN(X[[i]], ...) : only defined on a data frame with all numeric variables

### ISSUE THREAD

**Issue by weisskatze** [Microsoft/LightGBM#1851]
## Environment info

Operating System: Windows 10

CPU/GPU model: Ryzen 3 2200g

C++/Python/R version: R 3.5.1

## Error message

Error in FUN(X[[i]], ...) : only defined on a data frame with all numeric variables

## Reproducible examples

grid_search <- expand.grid(Depth = 8,
                           L1 = 0:5,
                           L2 = 0:5)

model <- list()
perf <- numeric(nrow(grid_search))

for (i in 1:nrow(grid_search)) {
  model[[i]] <- lgb.train(list(objective = "tweedie",
                          metric = "l2_root",
                           lambda_l1 = grid_search[i, "L1"],
                          lambda_l2 = grid_search[i, "L2"],
                          max_depth = grid_search[i, "Depth"]),
                          lgb.train,
                          2,
                          valids,
                          min_data = 1,
                          learning_rate = 1,
                          early_stopping_rounds = 5)
    perf[i] <- min(rbindlist(model[[i]]$record_evals$test$l2))
}

## Steps to reproduce

 Although 'grid_search' has only numeric values, why does this error occur?

@Laurae2
@BruceZhaoR

**Comment by guolinke** [comment]
ping @Laurae2

**Comment by Laurae2** [comment]
@weisskatze where does the issue occurs exactly?

I can't reproduce the issue.

```r
library(lightgbm)

data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
data(agaricus.test, package = "lightgbm")
test <- agaricus.test
dtest <- lgb.Dataset.create.valid(dtrain, test$data, label = test$label)
valids <- list(test = dtest)

i <- 1
grid_search <- expand.grid(Depth = 8, L1 = 0:5, L2 = 0:5)
model <- lgb.train(list(objective = "tweedie",
                        metric = "l2_root",
                        lambda_l1 = grid_search[i, "L1"],
                        lambda_l2 = grid_search[i, "L2"],
                        max_depth = grid_search[i, "Depth"]),
                   dtrain,
                   100,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10)
```

**Comment by StrikerRUS** [comment]
Closing due to inactivity.
Feel free to reopen with more details.
